src/Viewer.tsx

```diff
@@ -79,7 +79,11 @@ function Viewer(): ReactElement {
   const [datasetKey, setDatasetKey] = useState("");
   const [, addRecentCollection] = useRecentCollections();
 
-  const [featureKey, setFeatureKey] = useState("");
+  // TODO: Because feature data is cached, it's possible for the data to fall out of the cache and for the texture to get disposed of.
+  // Make the cache infinitely large? Locks? Ref counting?
+  // Make ColorizeCanvas request the texture data every time it renders?
+  const [featureData, setFeatureData] = useState<FeatureData | null>(null);
```

Comment:
Many of the changes in this file are caused by this line! Before we were storing `featureKey` and looking up the data from the dataset using `dataset?.getFeatureData(featureKey);`. However, `getFeatureData` is now async, so we store the results (`featureData`) instead of the key.